### DISCUSSION TITLE
Spring Cloud Stream - SQS & SNS Binder

### DISCUSSION THREAD

**Discussion by beccagaspard** [awspring/spring-cloud-aws#1160] [Ideas]
Any plans to add support for spring cloud stream binders for SQS & SNS?  ☁️ 

The repos for [spring-cloud-stream-binder-sqs](https://github.com/idealo/spring-cloud-stream-binder-sqs) and [spring-cloud-stream-binder-sns](https://github.com/idealo/spring-cloud-stream-binder-sns) that are linked from the [Spring Cloud Stream docs](https://spring.io/projects/spring-cloud-stream) have recently been archived and are no longer maintained. 

Looks like the archived binders were already leveraging spring-cloud-aws so would be great if they could somehow be reincarnated under the awspring umbrella!

**Comment by maciejwalkowiak** [comment]
@beccagaspard thanks for suggestion. Would you be interested in maintaining these modules under Spring Cloud AWS umbrella? :-)

**Reply by birbirsbirbirs** [reply]
I would be happy to maintain under some body supervision. Thank you very much.

**Reply by orvyl** [reply]
Just in time :) . Was searching for this for my service.

**Reply by backnight** [reply]
Would be very nice if official SNS/SQS binders would be available and maintained again.

So far I checked what the archived ones are missing:
- No handling of errors / ErrorMessage so there is no way of handling delivery fails, retries, back pressure, etc (Producers, especially for SNS)
- No support for batch sending (Producers, especially for SNS)
